### PR TITLE
perf(flake): remove runtime fetch operations

### PR DESCRIPTION
**Pull request by trueNAHO**
Closes: https://github.com/danth/stylix/issues/222

Additionally, I took the luxury to sort the `inputs` in `flake.nix`.

### CODE REVIEW

**Review by jalil-salame — APPROVED** [review]
LGTM

File: modules/gnome/theme.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ pkgs, config, ... }:
+{ config, inputs, pkgs, ... }:
```
Comment (danth) [inline modules/gnome/theme.nix:1]: I'm not sure `inputs` is defined here. The existing flake inputs are loaded via `stylix/templates.nix`

File: flake.nix [hunk 2]
```diff
@@ -1,52 +1,64 @@
...
+    flake-compat = {
       flake = false;
+      url = "github:edolstra/flake-compat";
     };
-    base16-vim = {
-      url = "github:chriskempson/base16-vim";
+
+    gnome-shell = {
       flake = false;
+      url = "git+https://gitlab.gnome.org/GNOME/gnome-shell";
```
[the remark this answers is not in the document]
Comment (trueNAHO) [reply flake.nix:52]: Considering that it is rather impossible to properly unit or integration test the GNOME shell we should simply rely on users reporting issues on version incrementations, and revert or adapt the patch in those cases. In all other case, which should be the majority of the time, upgrading has only benefits, including additional features, improved performance, and security patches.

I propose we unlock the version (although `flake.lock` locks it) and properly update versions until users report issues.

For reference, the commit unlocking the GNOME shell version is: https://github.com/danth/stylix/pull/224/commits/026bc2a2032ae309e0c3bfcf8c0b3ecb062a20c5.

File: modules/gnome/theme.nix (hunk 1 above)
Comment (trueNAHO) [reply modules/gnome/theme.nix:1]: `stylix/templates.nix` is used in the following contexts to extract certain attributes from `inputs`:

- https://github.com/danth/stylix/blob/a9e3ce064a778b386fb88fb152c02ae95aa2cbd2/stylix/darwin/default.nix#L14
- https://github.com/danth/stylix/blob/a9e3ce064a778b386fb88fb152c02ae95aa2cbd2/stylix/hm/default.nix#L15
- https://github.com/danth/stylix/blob/a9e3ce064a778b386fb88fb152c02ae95aa2cbd2/stylix/nixos/default.nix#L16

`stylix/templates.nix` contains the following:

https://github.com/danth/stylix/blob/a9e3ce064a778b386fb88fb152c02ae95aa2cbd2/stylix/templates.nix#L1-L15

To reduce the maintenance burden of manually adding every `inputs` entry to `stylix/templates.nix`, I suggest propagating the entire `inputs` set. In this case, the `stylix/templates.nix` file is obsolete and we replace the propagated `base-*` inputs to `inputs.base-*`.

Should we do that in this PR or a followup PR?

Comment (danth) [reply modules/gnome/theme.nix:1]: I agree, I don't see any reason to maintain a separate list.

Having it in `config.lib.stylix.templates` also means it gets exported, so we could end up with external code relying on our flake inputs.

Splitting into a followup PR would be better, but we'd need this PR to be in a state where everything is functional.

File: flake.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (trueNAHO) [reply flake.nix]: Actually, I locked the `gnome-shell` input in https://github.com/danth/stylix/pull/224/commits/5f4ea2cafb26bcaf5105dfc3caa280b849cb61ad.

File: modules/gnome/theme.nix (hunk 1 above)
Comment (trueNAHO) [reply modules/gnome/theme.nix]: Related: https://github.com/danth/stylix/pull/224/commits/dfdc18d8bdce94a4925d8aa3bf78c9c5a79de860

Comment (trueNAHO) [reply modules/gnome/theme.nix]: > I'm not sure `inputs` is defined here. The existing flake inputs are loaded via `stylix/templates.nix`

https://github.com/danth/stylix/pull/224/commits/dfdc18d8bdce94a4925d8aa3bf78c9c5a79de860 adds the `gnome-shell` input.

File: flake.nix [hunk 3]
```diff
@@ -1,52 +1,68 @@
...
-    base16-vim = {
-      url = "github:chriskempson/base16-vim";
+
+    gnome-shell = {
       flake = false;
+
+      # TODO: Unlocking the input and pointing to official repository requires
+      # updating the patch:
+      # https://github.com/danth/stylix/pull/224#discussion_r1460339607.
+      url = "github:GNOME/gnome-shell/45.1";
```
Comment (danth) [inline flake.nix:56]: We should use GitLab here since it's the original source (the GitHub version is a mirror).

Comment (trueNAHO) [reply flake.nix:56]: I am very well aware of the GitHub repository being a GitLab mirror: https://github.com/trueNAHO/stylix/commit/026bc2a2032ae309e0c3bfcf8c0b3ecb062a20c5. However, I do not know how to specify the revision for GitLab repositories. Note that `gitlab:GNOME/gnome-shell/45.1` does not work.

Reference: https://github.com/tweag/rfcs/blob/ca2845b81814c345945de4be0b36cd5fb4eb080b/rfcs/0049-flakes.md

**Review by danth — APPROVED** [review]

**Pull request merged**